hugr-py/src/hugr/function.py

```diff
@@ -57,3 +57,8 @@ def declare_function(self, name: str, signature: PolyFuncType) -> Node:
     def add_alias_decl(self, name: str, bound: TypeBound) -> Node:
         """Add a type alias declaration."""
         return self.hugr.add_node(ops.AliasDecl(name, bound), self.hugr.root)
+
+    @property
+    def metadata(self) -> dict[str, object]:
```

Comment:
`Module` is the only builder that doesn't implement `ToNode`, so it needs it's own metadata getter